Update searching.md

brishtibheja opened the pull request.

brishtibheja commented:
@dae It seems there was something called custom data the FSRS custom scheduling version used? Is that still needed? If not, searches for that should be removed. If other schedulers can still potentially use that, then people should be explained what is custom data. The manual seems to mention the phrase only thrice.

Danika-Dakika commented:
I have some suggested corrections and edits for this PR. I'm going to attempt to add them as comments on the changes themselves. We'll see if that makes it clearer than doing it from here.

Danika-Dakika commented on src/searching.md:
```diff
@@ -3,17 +3,16 @@
 
 Anki's Browse screen and the Filtered Deck feature use a common method
-of searching for specific cards/notes.
+of searching for specific cards/notes. This method is also used in the search bar below "FSRS parameters" field.
```
At a minimum this needs an article -- below ***the*** "FSRS ... 

But I don't think it's quite fitting to call that a "search bar," because that suggests it returns a result to the user. I would suggest: 
`This method can also be used to adjust the scope of FSRS optimization.`

Danika-Dakika replied:
> Filtered decks don't give you a search result

They return a set of cards in the deck based on the filter, so I'd say they do give the user a search result. 

> do we really need to be technically correct?

Yes, that's pretty much the entire point of the manual.

brishtibheja replied:
We are already using the word in the manual and in the FSRS tutorial. I added the 'bar' but I think it shouldn't be an issue.

brishtibheja replied:
```suggestion
of searching for specific cards/notes.  This method can also be used to adjust the scope of FSRS optimization. 
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -3,17 +3,16 @@
 and displays their cards. Anki searches in all fields of the notes, but
-does not search for tags (see later in this section for how to search
-for tags). Some examples:
+does not search for tags (For information on how to search for tags, see [this section](##tags-decks-cards-and-notes) of the page). Some examples:
```
Needs corrections to capitalization and/or punctuation (and the syntax of the link, I think). 

If the text is to be updated, in the manual, "section" is mostly used to refer to Summary/TOC items, and "page" usually refers to another site (like the FAQ). It's defnitely not 100%, but it would be better to stick with it and move toward consistency and coherency. 

But I don't think it is necessary to change the line at all to add this link -- just use the current text: 
`does not search for tags (see [later in this section](#tags-decks-cards-and-notes) for how to search for tags).`

brishtibheja replied:
```suggestion
does not search for tags (see [later in this section](#tags-decks-cards-and-notes) to search for tags). Some examples:
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -35,24 +34,24 @@ finds notes with neither "cat" nor "mouse".
 
 `-"a dog"`\
-finds notes without the exact phrase "a dog"
+finds notes without the following sequence of characters: "a dog".
```
Since this is offered in contrast to the example above, better to mirror that line: 
`finds notes without the exact sequence of characters "a dog" in them.`

brishtibheja replied:
I have mentioned below that some of this is intentional. This section/page of the manual has a lot of similar sounding sentences and it can quickly get boring to read through this. A little bit of variability, that catches the reader off guard, can draw their attention to what is written in the doc. I do value being consistent in formatting and else, but some amount of entropy is nice to have.

Danika-Dakika replied:
> some amount of entropy is nice to have.

In the manual, or in any technical writing? -- I strongly disagree. This isn't prose. Parallel phrasing is how you quickly and efficiently communicate similarity and subtle differences between things. 

Adding variability makes things harder for a reader to understand at-a-glance, and ruins the effect of contrasting the examples in the first place. Different wording/terminology should be used to signal for a reader that the manual is talking about about something different -- because that's the impression it gives.

brishtibheja replied:
```suggestion
finds notes without the exact sequence of characters "a dog" in them.
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -35,24 +34,24 @@ finds notes with neither "cat" nor "mouse".
-or "underdog". Requires Anki 2.1.24+ or AnkiMobile 2.1.61+. Note that
-formatting changes may be interpreted as word boundaries e.g. searching
-for `w:exam` will match **exam**ple.
+searches for the word "dog" as opposed to a sequence of characters - will match "dog", but not "doggy"
```
(Comment about "searches" above.)

If "word boundary" needs to be defined, it seems better to do that without removing the phrase itself, because it is somewhat of a term of art in search syntax. 
`search for "dog" on a word boundary (as a whole word, not a sequence of characters in another word) - will match`

brishtibheja replied:
In line 53 we use the term "word boundaries" and later we slowly transition to more frequent use of this. I don't think we really need to define word boundaries here. It is bit hard to define, and I think translators will find this sentence easier to deal with than your suggestion.

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
-find notes where the "Animal Front" field is exactly "a dog". The double quotes are
-mandatory, see [below](#matching-special-characters).
+finds notes where the "Animal Front" field is exactly "a dog". The double quotes are
+mandatory, see [this section](#matching-special-characters) of the page.
```
This line is fine as-is. If it needs clarification, as mentioned above, it should be consistent with uses of "section" and "page" in the manual.

brishtibheja replied:
```suggestion
mandatory, see [later in this section](#matching-special-characters).
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 
 `tag:ani*`\
-find notes with tags starting with ani
+finds notes with tags starting with ani.
```
Should add quotation marks as well, to match the other examples:
`finds notes with tags starting with "ani".`

brishtibheja replied:
```suggestion
finds notes with tags starting with "ani".
```
You can add the suggestions like this.

Danika-Dakika replied:
👍 I'll try to remember that for next time.

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 
 `deck:french`\
-find cards in a French deck, or subdecks like French::Vocab
+find cards in a deck called "French", or its subdecks like French::Vocab.
```
Should add quotation marks to subdeck name as well, to match the other examples:
`or its subdecks like "French::Vocab".`

brishtibheja replied:
```suggestion
find cards in a deck called "French", or its subdecks like "French::Vocab".
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 
 `"deck:french vocab"`\
-also ok
+same as earlier.
```
Since it is about physical space "earlier" doesn't fit here. 
Alternative A:
`same as above.`
Alternative B: combine the examples with one label for both. 
```
`deck:"french vocab"`\
`"deck:french vocab"`\
searching when the deck name contains a space.
```

brishtibheja replied:
Google uses "earlier". Referance: https://developers.google.com/style/word-list#above

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 `preset:"Default"`\
 cards in all decks that use the 'Default' deck options preset.
-Requires Anki 23.10+.
+Requires Anki(Mobile) 23.10+, or AnkiDroid 2.17+.
```
I haven't seen anywhere else that Anki and AnkiMobile are collapsed into a single name like that. I don't think it makes it clear what we're talking about. 

The full list would be better:
`Requires Anki 23.10+, AnkiMobile 23.10+, or AnkiDroid 2.17+.`

I don't think it needs to be more succinct, but if nececessary, another option would be:
`Requires Anki or AnkiMobile 23.10+, or AnkiDroid 2.17+.`

brishtibheja replied:
I think I missed updating some `Anki 23.10+` to `Anki(Mobile) 23.10+` but now I'm not sure whether that should be done. Everytime AnkiMobile is mentioned the version number is usually different from Anki. @dae how should this case be handled?

brishtibheja replied:
```suggestion
Requires Anki 23.10+, AnkiMobile 23.10+ or AnkiDroid 2.17+.
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 
 `deck:french -deck:french::*`\
-find cards in French, but not subdecks
+finds cards in French, but not subdecks.
```
Should add quotation marks as well, to match the other examples:
`finds cards in "French", but not its subdecks.`

brishtibheja replied:
```suggestion
finds cards in "French", but not its subdecks.
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 
 `card:forward`\
-search for Forward cards
+searches for cards with card type named "Forward".
```
I think this one, along with 160 and 164, should be changed from "search/es for" to "finds" -- to match the other examples. 

It sounds a tiny bit off to say "cards with card type named "Forward"." That phrasing sounds like a card has multiple types, like we might talk about a note "with" certain card types, not like card type is an inherent property of the card.

There are different alternatives for this, but I can't necessarily explain why they sound better to me. 
`cards of card type "Forward".`
`cards that are card type "Forward".`
`cards with card type "Forward".` [Interestingly, maybe it's just the "named" that throws it off, because this sounds fine.]
`cards with the card type of "Forward".`

brishtibheja replied:
I think if we don't include the word "named" people might think there are card types with an inherent property that makes them "Forward" card types. How about this:

```suggestion
searches for the cards of a card type named "Forward".
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 
 `note:basic`\
-search for cards with a Basic note type
+searches for cards with a note type named "Basic".
```
This has the same issue as above -- it's just a little bit off, but for a slightly different reason.

Some other options:
cards with the note type "Basic".
cards created by note type "Basic".

brishtibheja replied:
```suggestion
searches for cards created with a note type named "Basic".
```

I'm using "with" because it is the user who will be creating these cards.

Danika-Dakika replied:
I am not sure that clarifies things, especially since the reasoning runs counter to one of the primary concepts the manual needs to emphasize for users. 
> ... in Anki we add notes rather than cards, and Anki creates cards for us.

https://docs.ankiweb.net/editing.html#adding-cards-and-notes

Danika-Dakika commented on src/searching.md:
```diff
@@ -95,86 +94,86 @@ Things to note from the above:
 
 ## Ignoring accents/combining characters
 
 Requires Anki 2.1.24+ or AnkiMobile 2.0.60+.
```
Is there an AnkiDroid version that could be added here to complete the page?

brishtibheja replied:
```suggestion
Requires Anki 2.1.24+, AnkiMobile 2.0.60+ or AnkiDroid 2.17+.
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -183,148 +182,151 @@ Searches that ignore combining characters are slower than regular searches.
 a standard and powerful way of searching in text.
 
-Start a search with `re:` to search by regular expression. To make things easier, Anki will
+Start a search with `re:` to search using regular expression. To make things easier, Anki will
```
If changing "by" to "using," the plural sounds more accurate. I can't explain why. 
`search using regular expressions.`

brishtibheja replied:
```suggestion
Start a search with `re:` to search using regular expressions. To make things easier, Anki will
```

I agree with you.

Danika-Dakika commented on src/searching.md:
```diff
@@ -183,148 +182,151 @@ Searches that ignore combining characters are slower than regular searches.
+same as previous, but will not match if any other text falls before or after a1/b1/c1.
 
-Anki 2.1.50 added regex support for tags:
+Anki 2.1.50+ supports regular expressions for tags:
```
Are there AnkiMobile and AnkiDroid versions that could be added here?

brishtibheja replied:
AnkiDroid 2.17 probably, I'm not sure about AnkiMobile. @dae?

dae replied:
I'm not sure this is exposed by AnkiMobile yet.

Danika-Dakika commented on src/searching.md:
```diff
@@ -183,148 +182,151 @@ Searches that ignore combining characters are slower than regular searches.
 
 `is:suspended`\
-cards that have been manually suspended
+cards that have been automatically or manually suspended.
```
Can this line and the next (241) match? Either phrasing is fine.

brishtibheja replied:
I was about to follow your advice here, but I remembered that some of the variability is intentional. It is a bit boring to read a text if everything seems to be using some sort of parallel syntax. I don't think this should be changed.

Danika-Dakika replied:
Responded above at line 41.

brishtibheja replied:
I have resolved the other conversation as the explanation itself was confusing. For this one, I want to ask @dae why the original phrasing was different for is:suspended and is:burued? The docs are created in your vision so I'd like to get your feedback first.

dae replied:
I can't remember any specific reason for the differing wording, and don't have strong feelings here.

Danika-Dakika commented on src/searching.md:
```diff
@@ -183,148 +182,151 @@ Searches that ignore combining characters are slower than regular searches.
+manually.
 
-Note that with the [new scheduler](https://faqs.ankiweb.net/the-anki-2.1-scheduler.html),
+Note that with the [v2 scheduler](https://faqs.ankiweb.net/the-anki-2.1-scheduler.html),
```
I think it's great to catch this sentence up with the times! Unfortunately, it now sort of sounds like that is only a feature of the (no longer supported) v2 scheduler, and doesn't necessarily carry forward to v3. 

`Note that as of the [v2 scheduler](https://faqs.ankiweb.net/the-anki-2.1-scheduler.html), Anki distinguishes`
`Note that starting with the [v2 scheduler](https://faqs.ankiweb.net/the-anki-2.1-scheduler.html), Anki distinguishes`

[On the other hand, this is a very strange place for this information -- a note about unburying in the middle of a section about search syntax. It might be better to move this to wherever unburying is documented.]

Danika-Dakika commented on src/searching.md:
```diff
@@ -183,148 +182,151 @@ Searches that ignore combining characters are slower than regular searches.
-`prop:cdn:d>5` (Requires Anki 2.1.64+.)\
-cards with the value of `d` in custom data (usually refers to Difficulty in FSRS) greater than 5
+`prop:cdn:d>5`\
+cards with the value of `d` in custom data (usually refers to difficulty in FSRS) greater than 5 (requires Anki 2.1.64+).
```
I think capitalizing "Difficulty" is correct here, since it's referring to one of the titled values of FSRS.

brishtibheja replied:
We don't capitalise DSR in the statistics screen of app. Here is the part of Sherlock's essay where he doesn't capitalise these terms: [Spaced Repetition Algorithm](https://github.com/open-spaced-repetition/fsrs4anki/wiki/Spaced-Repetition-Algorithm:-A-Three%E2%80%90Day-Journey-from-Novice-to-Expert#:~:text=Below%20are%20some%20empirical%20observations%20from%20Wozniak%2C%20corroborated%20by%20data%20from%20language%20learning%20platforms%20like%20MaiMemo%3A)

Danika-Dakika commented on src/searching.md:
```diff
@@ -335,47 +337,46 @@ included even if their notes were added a long time ago.
-special meaning in Anki. If you need to locate those characters in a search,
-you need to tell Anki not to treat them specially.
+special meaning in search. If you need to locate those characters in a search,
+you need to tell Anki not to treat them specially. This is called escaping a character and is primarily done by using double quotes and backslashes.
```
`"escaping" `
or
`"escaping a character"`

brishtibheja replied:
```suggestion
you need to tell Anki not to treat them specially. This is called "escaping a character" and is primarily done by using double quotes and backslashes.
```

Danika-Dakika commented on src/searching.md:
```diff
@@ -387,27 +388,26 @@ you need to tell Anki not to treat them specially.
-  `"some\(text\)"` are all equivalent, but `some(text)` is not.
+  You can search for parentheses by enclosing the entire term in quotes,
+   by using a backslash, or both at the same time. For example, `"(text)"`, `\(text\)` and
+  `"\(text\)"` are all equivalent searches, and searches for `(text)`.
```
At the end, this should be:
`and search for `(text)`.`

But I don't agree with removing the more detailed examples that include "some" (unless they were incorrect in some way I'm not seeing?). I think they illustrate well that the entire term needs to be enclosed.

brishtibheja replied:
```suggestion
  `"\(text\)"` are all equivalent searches, and search for `(text)`.
```
I believe the reader, if he has read other parts of this page/section will already understand what 'entire term' means. A simpler example is better here.

Danika-Dakika commented on src/searching.md:
```diff
@@ -387,27 +388,26 @@ you need to tell Anki not to treat them specially.
-  `w:e:b` and `w\:e:b` searches the field `w:e` for `b` (see
-  [field searches](#limiting-to-a-field)).
+  Colons have to be escaped using backslashes unless they are preceded by another, unescaped colon.
+  For example, `w:3:30` searches for 3:30 on word boundary and doesn't require you to use a backslash.
```
Should use quotation marks, to match other examples:
`searches for "3:30" on word boundary`

brishtibheja replied:
```suggestion
  For example, `w:3:30` searches for "3:30" on word boundary and doesn't require you to use a backslash.
```

brishtibheja commented on src/searching.md:
```diff
@@ -3,17 +3,16 @@
 
 `dog`\
-search for "dog" - will match words like "doggy" and "underdog" too.
+searches for "dog" - will match words like "doggy" and "underdog" too.
```
The conjugation will feel consistent otherwise. In the very next phrase we say "will match words" as if to say "the search term will match words like...". It is only appropriate to use searches instead of search, to mean "this term searches for 'dog'. Using  "search" will make it sound like an imperative sentence, as in, "Do this" or "Search that", or a TV ad "use our new...".

Danika-Dakika replied:
I see how that's another fair reading of it.

brishtibheja commented:
@Danika-Dakika I have gone through all suggestions (I wished you added the suggestions in code block so I could just commit the changes directly; there's a button to do that when you comment), I have most of them resolved, otherwise I added comments. I have added a few extra commits too, probably the last ones. Hopefully, this will get merged when Dae comes. On a different note, are there sections of manual you deem important as a helper that haven't been updated in a while or is difficult-to-understand, etc.?

Danika-Dakika commented:
> On a different note, are there sections of manual you deem important as a helper that haven't been updated in a while or is difficult-to-understand, etc.?

It's an interesting question, but I don't want to think it through in this PR, because we'll never find it again. 😅 
I have a related idea that I'm going to post in the Forums, and it might lead to an answer for this!

dae commented on src/searching.md:
```diff
@@ -183,148 +182,155 @@ Searches that ignore combining characters are slower than regular searches.
+all cards due in the future, including tomorrow.
+
+`prop:due<-1`\
+all cards due in the past (that haven't been answered yet), including yesterday.
```
```suggestion
all  overdue cards
```

dae reviewed (commented):
Thank you both for your hard work on this!

dae commented:
`Custom data` allows storing extra data in cards. The old FSRS implementation used it, but it was not FSRS-specific, and may be used in niche other use-cases, but is probably obscure enough that we can just stop documenting it for now.

brishtibheja commented on src/searching.md:
```diff
@@ -183,148 +182,155 @@ Searches that ignore combining characters are slower than regular searches.
 
 `front:re:^[a-c]1$`\
-like the above, but will not match if any other text falls before or after a1/b1/c1.
+same as previous, but will not match if any other text falls before or after a1/b1/c1.
```
```suggestion
same as the previous example, but will not match if any other text falls before or after a1/b1/c1.
```

brishtibheja commented:
@dae I believe this can be merged. I have also added documentation for escaping "and" and "or" so the other PR can be closed.

dae commented on src/searching.md:
```diff
@@ -376,12 +376,16 @@ If you're using a version earlier than Anki 2.1.36 the following searches may no
 special meaning in search. If you need to locate those characters in a search,
 you need to tell Anki not to treat them specially. This is called "escaping a character" and is primarily done by using double quotes and backslashes.
-
+  
```
Intentional?

brishtibheja replied:
It's not.

dae replied:
```suggestion

```

Danika-Dakika commented on src/searching.md:
```diff
@@ -183,148 +182,155 @@ Searches that ignore combining characters are slower than regular searches.
-cards due yesterday that haven’t been answered yet
+cards due yesterday that haven’t been answered yet.
+
+`prop:due>1`\
+all cards due in the future, including tomorrow.
+
+`prop:due<-1`\
+all  overdue cards
```
This just caught my eye as I was catching up on comments. 

```suggestion
`prop:due>=1`\
all cards due in the future, including tomorrow.

`prop:due<=-1`\
all  overdue cards
```

dae commented:
Thank you both!
